### PR TITLE
Autolathe economy overhaul v2

### PR DESCRIPTION
**Pull request by mudzbe**
## About The Pull Request

I am back after a short break... I have this amazing PR for you. I have massively changed this since it's last iteration. What has been added?

- Printing items from the autolathe costs money depending on the amount and type of mats it uses to print
- An ID or department budget can be linked to the autolathe. 70% of the fee will go to the linked account, 10% to the civilian budget and 20% disappears to NT, the syndicate or maybe somewhere else? Your guess is as good as mine. 
- The linked account can be cleared by pulsing a new wire.
- The autolathe can be set to "free mode" where everything is free, if the autolathe spawns/is built offstation, it will be set automatically to this.
- The autolathe can be emagged to be set to free mode.
- All autolathe item's material value have been changed to properly balance the costs.

Mat costs/prices have been set to reflect their individual value rather than realistic mineral cost. Generally if an item can already be bought at a vendor, it is cheaper at that vendor by a factor of 2.5. Basic tools like crowbars and screwdrivers have been set to 50c, multitool is 67c, welding tools are around that price too. I haven't changed part mat requirements and as such they are cheaper, ranging around 20c for basic parts. Weapons and ammo are more expensive, shotgun shells range at 50c. All of these prices are negotiable and I don't expect them to be perfect, even so, insignificant pricing is better than no pricing.

## Why It's Good For The Game

This adds another avenue of balancing items rather than is there enough mats yes/no? I saw some people wanted to change prices in vendors and this supports that ability by not having a free source of items that is objectively better than those vendors.
Money has long been near useless and I aim to change that.  Fining people as a means of punishment has more weight now.

What about assistants, they have no money? They wont be able to buy anything! Good. Those lazy parasites should do something productive instead of getting free handouts and committing crimes with those handouts. Next, if this is merged I will add pricing to items from protolathes, that too is a case of is it researched Y/N? Is there enough mats Y/N? I can also tie in the ability for the HOP being able to change the percentage of tax to the civ budget and for the price to inflate because of that, hooray.

Let's talk stats. I counted 45 possible sets of tools roundstart on boxstation.
- 10 sets were in public areas.
- 10 sets purchasable from youtools.
- 20 sets were in restricted areas, these can be negotiated for.
- 5 misc sets of tools spread around both.
- any I missed

There is an abundance of tools roundstart and if all else fails money can be gained by:
- 24 bounties + 5 extra bounties for enterprising assistants. Any of which can buy several sets of tools from an autolathe several times over.
- Every crew member can be convinced through a loan, an odd job, or something else creative.
- 7 department budgets with an abundance of money, talk to the heads
- Assistants can pitch in together and share a set of tools

Later on into the round the autolathe would be upgraded and tools printed from there will be cheaper than purchasing them from youtools, youtools can also be restocked. Simply put assistants getting tools wont be an issue if they make an effort. Why should assistants get handouts if they disproportionately commit more crime?

## Changelog
add: Autolathe items cost money
add: Accounts can be linked to autolathes
add: New wire for the autolathe
add: Autolathe freemode for communists and ghost roles
add: Emagged autolathes enter freemode
balance: rebalanced mat costs for autolathe items
/:cl:

### CODE REVIEW

**Comment by PowerfulBacon** [comment]
> Basic tools like crowbars and screwdrivers have been set to 50c, multitool is 67c, welding tools are around that 

You cant do shit with the economy until you lower the pay gap between assistants and command. This is worthless to well paid jobs, but about 40 minutes for 1 screwdriver for assistants. The only way I can see this not being awful is if there was a hacking wire to enable free mode, but at that point this PR is nothing but an inconvenience and has no point

**Comment by PowerfulBacon** [comment]
This is one of those nerf PRs that fixes the problem it aims to do without considering the massive impact it has on the things that are good.

**Comment by eeSPee** [comment]
I mean, I get the spirit, but we already have vending machines that sell tools and components that are made by autolathes.

Why not make a PR that:
* Puts autolathes behind closed doors
* Adds one of them vendors in cargo and science
* Increases assistant salaries

**Comment by eeSPee** [comment]
> > Basic tools like crowbars and screwdrivers have been set to 50c, multitool is 67c, welding tools are around that
> 
> You cant do shit with the economy until you lower the pay gap between assistants and command. This is worthless to well paid jobs, but about 40 minutes for 1 screwdriver for assistants. The only way I can see this not being awful is if there was a hacking wire to enable free mode, but at that point this PR is nothing but an inconvenience and has no point

From the wiki:

>**Assistants are expected to be useful.**
>* You do not have a free pass to do your own thing. This includes wandering maintenance for no reason or in search of gear/loot.

**Comment by Rukofamicom** [comment]
> things that are good.

Insinuating free public tool access is good. Public autolathes were a mistake in the first place, and so was adding multitools to Youtools. 

Tiderstation13 is dumb. 

> What about assistants, they have no money? They wont be able to buy anything! Good. Those lazy parasites should do something productive instead of getting free handouts and committing crimes with those handouts.

Assistants are:
* for new players that should need to ask for help 
* for experienced players that wish to fill in wherever they are sent
* **not** for treating like a gimmick role that fucks off and does whatever it wants. We have dedicated gimmick roles that are heavily limited for a reason.

**Comment by PowerfulBacon** [comment]
> should do something productive

Like what? Fix maint wires? repair breaches? Build a shop / bar in maints?

The only thing productive you can do as assistant without tools is... uhhh nothing really.

Making them not public is a 100x simpler solution and does essentially the same thing without the huge problems like having a small effect on people riding while having a large effect on anyone wanting to print a lot (the people doing good builds like cargo making conveyors and stuff like that). A solution shouldn't be worse than the problem.

**Comment by Cenrus** [comment]
I'd rather have autolathes be made private again.

**Comment by super12pl** [comment]
The fun
hand it over

**Comment by ToasterBan** [comment]
spend some bucks to print screwdriver and wirecutter

hack the free wire

print the rest of tools for free

**Comment by Effemu** [comment]
>spawn in as assistant because all jobs are already taken
>breaches everywhere
>uh oh

**Comment by Ohagi-Chan** [comment]
> spend some bucks to print screwdriver and wirecutter
> 
> hack the free wire
> 
> print the rest of tools for free

The new wire resets the linked account.
Autolathes can't be made free unless you're building off station or an admin.

File: code/game/machinery/autolathe.dm [hunk 1]
```diff
@@ -97,9 +107,15 @@
...
 
 /obj/machinery/autolathe/ui_static_data(mob/user)
-	var/list/data = list()
+	var/list/data = list()	
 	data["acceptsDisk"] = TRUE
 
+	var/static/list/ore_values = list(iron = 1, glass = 1, copper = 3, plasma = 15, silver = 16, gold = 18, titanium = 30, uranium = 30, diamond = 50, bluespace = 50, bananium = 60)
+
+	var/turf/here = get_turf(src)
+	if(!is_station_level(here.z))
```
Comment (Archanial) [inline code/game/machinery/autolathe.dm]:
```suggestion
	if(!is_station_level(z))
```

File: code/game/machinery/autolathe.dm [hunk 2]
```diff
@@ -450,10 +493,39 @@
...
-
 		materials_used[used_material] = amount_needed
 
+	var/datum/bank_account/B
+
+	if(free_mode)
+		price = 0
+	else
+		if(price < MINIMUM_PRICE) //To ensure the price isnt too low
+			price += MINIMUM_PRICE
```
Comment (Archanial) [inline code/game/machinery/autolathe.dm:503]:
```suggestion
			price = MINIMUM_PRICE
```

**Comment by mudzbe** [comment]
> spawn in as assistant because all jobs are already taken
> breaches everywhere
> uh oh

I have literally never seen an assistant fix a breach lol

> > spend some bucks to print screwdriver and wirecutter
> > hack the free wire
> > print the rest of tools for free
> 
> The new wire resets the linked account.
> Autolathes can't be made free unless you're building off station or an admin.

Built off station or emagged

> > should do something productive
> 
> Like what? Fix maint wires? repair breaches? Build a shop / bar in maints?
> 
> The only thing productive you can do as assistant without tools is... uhhh nothing really.
> 
> Making them not public is a 100x simpler solution and does essentially the same thing without the huge problems like having a small effect on people riding while having a large effect on anyone wanting to print a lot (the people doing good builds like cargo making conveyors and stuff like that). A solution shouldn't be worse than the problem.

There are plenty of ways assistants can get money, if they really want tools they should do atleast some work for it. Cargo bounties, asking for a loan from heads or people you agree to help. A single cargo bounty if cargo gave you even 50% of the cut would pay for their tools and then some. Let’s also not forget the toolboxes lying around, the tool vendors which are cheaper and the fact autolathes can be upgraded with parts. If you upgrade them with T4 I’m I think they’d be even cheaper than the vendors. You could always ask for money from random people too if you ain’t a shitter. Since money is so worthless now it means giving an assistant your whole wage for a simple task or for free isn’t unreasonable. I will reduce the assistant wage gap though definitely.

**Comment by mudzbe** [comment]
> I mean, I get the spirit, but we already have vending machines that sell tools and components that are made by autolathes.

I have seen atleast one PR that wanted to change vendor prices but couldn’t because the autolathe tools were free, I don’t know what happened to it but I think it got closed. Autolathes make other items that vendors don’t, vendor items were made intentionally more expensive with T1 parts to print in the autolathe to not make vendors redundant.

> Why not make a PR that:
> * Increases assistant salaries

I will lower the gap

**Comment by ToasterBan** [comment]
>buy screw driver and wirecutter
>hack owner wire
>become new owner
>get free tools

**Comment by PowerfulBacon** [comment]
#4033 is a better solution

**Comment by mudzbe** [comment]
TO MAKE IT CLEAR I DONT HATE ASSISTANTS OR GREYTIDE. I just want the economy to be more robust, it just happens I could meme on assistants as well

**Comment by PowerfulBacon** [comment]
Could port TGs economy, our economy is garbage at the moment and something like this is unfeasible. Assistants don't really get it worse off since it affects everyone negatively.

**Comment by Glcbitj** [comment]
You people really want assistants to start invading Sci/Engie every round for the free protolathe prints?

And if you do add a cost to the Proto's, that'll either make people beg for heads for Budget money or greytide into their lockers for the Budget card.

**Comment by mudzbe** [comment]
Let's talk stats. I counted 45 possible sets of tools roundstart on boxstation. 
- 10 sets were in public areas. 
- 10 sets purchasable from youtools.  
- 20 sets were in restricted areas. 
- 5 misc sets of tools spread around both. 
- + any I missed

There is an abundance of tools roundstart and if all else fails money can be gained by:
- 24 bounties + 5 extra bounties for enterprising assistants. Any of which can buy several sets of tools from an autolathe several times over.
- Every crew member can be convinced through a loan, an odd job, or something else creative.
- 7 department budgets with an abundance of money, talk to the heads
- Assistants can pitch in together and share a set of tools

Later on into the round the autolathe would be upgraded and tools printed from there will be cheaper than purchasing them from youtools, youtools can also be restocked. Simply put assistants getting tools wont be an issue.

File: code/game/machinery/autolathe.dm (hunk 2 above)
Comment (mudzbe) [reply code/game/machinery/autolathe.dm:503]: A matter of preference

File: code/game/machinery/autolathe.dm (hunk 1 above)
Comment (mudzbe) [reply code/game/machinery/autolathe.dm]: If it works

**Comment by mudzbe** [comment]
> > buy screw driver and wirecutter
> > hack owner wire
> > become new owner
> > get free tools

Not free, a 70% discount. But I will fix this by adding requiring two swipes, an alert to the registered account and a cooldown in-between the two swipes

**Comment by Ohagi-Chan** [comment]
> > > buy screw driver and wirecutter
> > > hack owner wire
> > > become new owner
> > > get free tools
> 
> Not free, a 70% discount. But I will fix this by adding requiring two swipes, an alert to the registered account and a cooldown in-between the two swipes

I can see this being worked around simply by disassembling the autolathe.
You can't really fix this because the board itself has a department pay flag that can be used to save the owner, but it doesn't take accounts to my knowledge.

Best you can do is probably notify the person holding a linked card, whenever their machine is unlinked, either through hacking or disassembly.

**Comment by mudzbe** [comment]
Im too busy speaking to girls

**Pull request closed without merging**